core/chain_indexer.go

```diff
@@ -219,21 +216,40 @@ func (c *ChainIndexer) eventLoop(currentHeader *types.WorkObject, events chan Ch
 				errc <- nil
 				return
 			}
-			header := ev.Block
+			select {
+			case qiIndexerCh <- ev.Block:
+			default:
+				c.logger.Warn("qiIndexerCh is full, dropping block")
+			}
+		}
+	}
+}
 
+func (c *ChainIndexer) indexerLoop(currentHeader *types.WorkObject, qiIndexerCh chan *types.WorkObject, nodeCtx int, config params.ChainConfig) {
+	var (
+		prevHeader = currentHeader
+		prevHash   = currentHeader.Hash()
+	)
+	for {
+		select {
+		case errc := <-c.quit:
+			// Chain indexer terminating, report no failure and abort
+			errc <- nil
```

Comment:
why do you swallow the error? wouldn't it be better to bubble it up?

Reply:
The code was copied from the eventLoop function above, but it's not actually an error, just mislabeled. The channel is only used to quit the indexer in the event of node shutdown.